src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -176,11 +181,19 @@ public X509Certificate[] getAcceptedIssuers() {
                 }
             }
 
-            // initialize truststore
-            KeyStore ts = KeyStore.getInstance(KeyStore.getDefaultType());
-            ts.load(null, null);
+            KeyStore ts = KeyStore.getInstance(truststoreType);
 
-            for (TruststoreConfig truststore : truststores) {
+            // initialize truststore with user provided path and pass
+            if (truststorePath.isEmpty() && truststorePass.isEmpty()) {
+                try (InputStream truststore = new FileInputStream(truststorePath.get())) {
+                    ts.load(truststore, truststorePass.get().toCharArray());
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            // initialize truststore with user provided certs
+            for (TruststoreConfig truststore : truststoreCerts) {
                 // load truststore with certificatesCertificate
                 InputStream certFile = new FileInputStream(truststore.getPath());
```

Comment:
I missed this the first time around - I think this could be done with autoclose, ie `try (InputStream certFile = ...)`, to be sure it's properly released.